dwave/system/samplers/clique.py

```diff
@@ -31,6 +33,127 @@
+    and quadratic terms accordingly.
+    Checks whether the per qubit coupling range is violated for the qpu 
+    and rescale accordingly. 
+
+    Args:
+       sampler (:obj:`dimod.Sampler`):
+            A dimod sampler.
+
+    This is local version of the parent class ScaleComposite, intended to be
+    used only with a DwaveCliqueSampler instance. 
+    """
+    @dimod.decorators.nonblocking_sample_method
+    def sample(self, bqm, scalar=None, bias_range=1, quadratic_range=None,
+               ignored_variables=None, ignored_interactions=None,
+               ignore_offset=False, **parameters):
+        """ Scale and sample from the provided binary quadratic model.
+
+        if scalar is not given, problem is scaled based on bias and quadratic
+        ranges. See :meth:`.BinaryQuadraticModel.scale` and
+        :meth:`.BinaryQuadraticModel.normalize`
+
+        Args:
+            bqm (:obj:`dimod.BinaryQuadraticModel`):
+                Binary quadratic model to be sampled from.
+
+            scalar (number):
+                Value by which to scale the energy range of the binary
+                quadratic model. Overrides `bias_range` and `quadratic_range`.
+
+            bias_range (number/pair, default=1):
+                Value/range by which to normalize the all the biases, or if
+                `quadratic_range` is provided, just the linear biases.
+                Overridden by `scalar`.
+
+            quadratic_range (number/pair):
+                Value/range by which to normalize the quadratic biases.
+                Overridden by `scalar`.
+
+            ignored_variables (iterable, optional):
+                Biases associated with these variables are not scaled.
+
+            ignored_interactions (iterable[tuple], optional):
+                As an iterable of 2-tuples. Biases associated with these
+                interactions are not scaled.
+
+            ignore_offset (bool, default=False):
+                If True, the offset is not scaled.
+
+            **parameters:
+                Parameters for the sampling method, specified by the child
+                sampler.
+
+        Returns:
+            :obj:`dimod.SampleSet`
+
+        """
+        original_bqm = bqm
+        bqm = bqm.copy()  # we're going to be scaling
+
+        if scalar is not None:
+            bqm.scale(scalar,
+                      ignored_variables=ignored_variables,
+                      ignored_interactions=ignored_interactions,
+                      ignore_offset=ignore_offset)
+        else:
+            scalar = bqm.normalize(bias_range, quadratic_range,
+                                   ignored_variables=ignored_variables,
+                                   ignored_interactions=ignored_interactions,
+                                   ignore_offset=ignore_offset)
+
+        if scalar == 0:
+            raise ValueError('scalar must be non-zero')
+
+        if 'per_qubit_coupling_range' in self.child.properties.keys():
```

Comment:
Given that all of the new logic happens _after_ the default scaling behaviour, maybe it makes sense to make an even more custom
```
class _QubitCouplingComposite(ComposedSampler):
    ....
```
and then later do
```
sampler = FixedEmbeddingComposite(
            ScaleComposite(
                _QubitCouplingComposite(
                    self.child)),
            embedding)

```
which would have the advantage of not needing to duplicate the `ScaleComposite` logic.